examples/customresources/demos/remoteserver/server_test.go

```diff
@@ -20,74 +21,117 @@ import (
 	"go.viam.com/rdk/logging"
 	robotimpl "go.viam.com/rdk/robot/impl"
 	weboptions "go.viam.com/rdk/robot/web/options"
+	"go.viam.com/rdk/testutils/robottestutils"
 	"go.viam.com/rdk/utils"
 )
 
 func TestGizmo(t *testing.T) {
-	port1, err := goutils.TryReserveRandomPort()
-	test.That(t, err, test.ShouldBeNil)
-	addr1 := fmt.Sprintf("localhost:%d", port1)
-	test.That(t, err, test.ShouldBeNil)
-	port2, err := goutils.TryReserveRandomPort()
-	test.That(t, err, test.ShouldBeNil)
-	addr2 := fmt.Sprintf("localhost:%d", port2)
-
+	// This test sets up three robots as a chain of remotes: MainPart -> A -> B. For setup, the test
+	// brings up the robots in reverse order. Remote "B" constructs a component using a custom
+	// "Gizmo" API + model. The test then asserts a connection to the `MainPart` can get a handle on
+	// a remote resource provided by "B". Additionally, remote "A" is not aware of the gizmo API nor
+	// the gizmo model.
 	ctx := context.Background()
 	logger := logging.NewTestLogger(t)
 
-	cfgServer, err := config.Read(ctx, utils.ResolveFile("./examples/customresources/demos/remoteserver/remote.json"), logger)
-	test.That(t, err, test.ShouldBeNil)
-	r0, err := robotimpl.New(ctx, cfgServer, logger.Sublogger("gizmo.server"))
-	test.That(t, err, test.ShouldBeNil)
-	defer func() {
-		test.That(t, r0.Close(context.Background()), test.ShouldBeNil)
-	}()
-	options := weboptions.New()
-	options.Network.BindAddress = addr1
-	err = r0.StartWeb(ctx, options)
```

Comment:
Strictly speaking, this part of this test wouldn't cause an issue like we see with the others. If the port was taken, `server.Start` would return the failure to bind error.

But I figured if I was adding loops around the other tests to be robust, might as well get this one was well.